grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcHealthCheckServiceBuilder.java

```diff
@@ -0,0 +1,96 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.grpc;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.healthcheck.HealthCheckUpdateListener;
+import com.linecorp.armeria.server.healthcheck.ListenableHealthChecker;
+
+/**
+ * Builds a {@link GrpcHealthCheckService}.
+ */
+@UnstableApi
+public final class GrpcHealthCheckServiceBuilder {
+
+    private final ImmutableSet.Builder<ListenableHealthChecker> serverHealthCheckers = ImmutableSet.builder();
+
+    private final ImmutableMap.Builder<String, ListenableHealthChecker> grpcHealthCheckers =
+            ImmutableMap.builder();
+
+    private final ImmutableList.Builder<HealthCheckUpdateListener> updateListeners = ImmutableList.builder();
+
+    GrpcHealthCheckServiceBuilder() {}
+
+    /**
+     * Adds the specified {@link ListenableHealthChecker}s that determine the healthiness of the {@link Server}.
+     */
+    public GrpcHealthCheckServiceBuilder checkers(ListenableHealthChecker... serverHealthCheckers) {
+        return checkers(ImmutableSet.copyOf(requireNonNull(serverHealthCheckers, "healthCheckers")));
+    }
+
+    /**
+     * Adds the specified {@link ListenableHealthChecker}s that determine the healthiness of the {@link Server}.
+     */
+    public GrpcHealthCheckServiceBuilder checkers(
+            Iterable<? extends ListenableHealthChecker> serverHealthCheckers) {
+        this.serverHealthCheckers.addAll(requireNonNull(serverHealthCheckers, "healthCheckers"));
+        return this;
+    }
+
+    /**
+     * Adds the pair of the service name and the {@link ListenableHealthChecker} that determine
+     * the healthiness of the specified gRPC service.
+     * The specified health checker is not used for determining the healthiness of the {@link Server}.
+     *
+     * <p>Note: The suggested format of service name is `package_names.ServiceName`.
+     * For example, If the proto package that defines `HelloService` belongs to `com.example`,
+     * the service name is `com.example.HelloService`
+     */
+    public GrpcHealthCheckServiceBuilder serviceCheckers(String serviceName,
+                                                         ListenableHealthChecker healthChecker) {
+        grpcHealthCheckers.put(requireNonNull(serviceName, "serviceName"),
+                               requireNonNull(healthChecker, "healthChecker"));
+        return this;
+    }
+
+    /**
+     * Adds a {@link HealthCheckUpdateListener} which is invoked when the healthiness of the {@link Server} is
+     * updated.
+     */
+    public GrpcHealthCheckServiceBuilder updateListener(HealthCheckUpdateListener updateListener) {
```

Comment:
Note: I find this API a little awkward since `updateListener`s don't receives callbacks for a specific service (as provided by `serviceChecker`.
Having said this, I don't think this is a big deal for now.